.circleci/config.yml

```diff
@@ -37,15 +37,43 @@ jobs:
       - run:
           name: Run regular tests
           command: |
-            CLASSNAMES=$(circleci tests glob "**/src/test/**/**.java" | grep -v 'spring-cloud-kubernetes-integration-tests' \
+            # find class names that do not have @Nested inside them
+            circleci tests glob "**/src/test/**/**.java" \
+                               | grep -v 'spring-cloud-kubernetes-integration-tests' \
+                               | grep -v '@Nested' \
                                | xargs grep -l '@Test' \
+                               | xargs grep -L '@Nested' \
                                | sed 's/.*src.test.java.//g' | sed 's@/@.@g' \
                                | sed 's/.\{5\}$//' \
-                               | circleci tests split --split-by=timings)
-            echo $CLASSNAMES
-            TEST_ARG=$(echo $CLASSNAMES | sed 's/ /,/g')
-            echo $TEST_ARG
-            ./mvnw -s .settings.xml  -DfailIfNoTests=false -DtestsToRun=$TEST_ARG -e clean org.jacoco:jacoco-maven-plugin:prepare-agent test -U -P sonar -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+                               > /tmp/all_tests.txt
+
+            # find classnames that have @Nested inside them
```

Comment:
the idea here is this:

we find "root" classes, that contain `@Nested` inside them; then go inside these classes and find all the inner classes that they have, basically this:

```
flattened=$(cat $p | awk '/@Nested/, /@Test/' | tr '\n' ' ' | awk '{ for (i = 1; i <= NF; ++i) { if( $i == "class") { print $(i+1) } } }')
```

once the inner classes are found, compute their fully qualified names. For example:

```
org.springframework.cloud.kubernetes.client.KubernetesClientActuatorTests$DisabledHealthTest
```